plugins/gcov/gcov.rake

```diff
@@ -154,67 +154,310 @@ if PROJECT_USE_DEEP_DEPENDENCIES
+      run_gcovr(args)
+    else
+      # gcovr version 4.1 and earlier supports HTML and Cobertura XML reports.
+      # It does not support SonarQube and JSON reports.
+      # Reports must also be generated separately.
+      args_cobertura = gcovr_args_builder_cobertura(opts, true)
+      args_html = gcovr_args_builder_html(opts, true)
+
+      if args_html.length > 0
+        print "Creating a gcov HTML report in '#{GCOV_ARTIFACTS_PATH}'... "
+        STDOUT.flush
+
+        # Generate the HTML report.
+        run_gcovr(args_common + args_html)
+      end
+
+      if args_cobertura.length > 0
+        print "Creating a gcov XML report in '#{GCOV_ARTIFACTS_PATH}'... "
+        STDOUT.flush
+
+        # Generate the Cobertura XML report.
+        run_gcovr(args_common + args_cobertura)
+      end
+    end
+
+    # Determine if the text report is enabled. Defaults to disabled.
+    if is_utility_enabled(opts, UTILITY_NAME_GCOVR) && is_report_type_enabled(opts, REPORT_TYPE_TEXT)
+      make_gcovr_text_report(opts, args_common)
+    end
+  end
+
+
+  desc "Create gcov code coverage html/xml/json/text report(s). (Note: Must run 'ceedling gcov' first)."
+  task GCOV_SYM do
+    # Get the gcov options from project.yml.
+    opts = @ceedling[:configurator].project_config_hash
+
+    # Create the artifacts output directory.
+    if !File.directory? GCOV_ARTIFACTS_PATH
+      FileUtils.mkdir_p GCOV_ARTIFACTS_PATH
+    end
+
+    # Remove unsupported reporting utilities.
+    if !(opts[:gcov_utilities].nil?)
+      opts[:gcov_utilities].reject! { |item| !([UTILITY_NAME_GCOVR].include? item) }
+    end
+
+    # Default to gcovr when no reporting utilities are specified.
+    if opts[:gcov_utilities].nil? || opts[:gcov_utilities].empty?
+      opts[:gcov_utilities] = [UTILITY_NAME_GCOVR]
+    end
+    
+    # Default to HTML basic report when no report types are defined.
+    if opts[:gcov_report_types].nil? && opts[:gcov_html_report_type].nil? && opts[:gcov_xml_report].nil?
+      opts[:gcov_report_types] = [REPORT_TYPE_HTML_BASIC]
+
+      puts "In your project.yml, define one or more of the"
+      puts "following to specify which reports to generate."
+      puts "For now, creating only an #{REPORT_TYPE_HTML_BASIC} report."
+      puts ""
+      puts ":gcov:"
+      puts "  :report_types:"
+      puts "    - #{REPORT_TYPE_HTML_BASIC}"
+      puts "    - #{REPORT_TYPE_HTML_DETAILED}"
+      puts "    - #{REPORT_TYPE_TEXT}"
+      puts "    - #{REPORT_TYPE_COBERTURA}"
+      puts "    - #{REPORT_TYPE_SONARQUBE}"
+      puts "    - #{REPORT_TYPE_JSON}"
+      puts ""
     end
 
-    if xml_enabled
-      puts "Creating an xml report of gcov results in #{GCOV_ARTIFACTS_FILE_XML}..."
-      command = @ceedling[:tool_executor].build_command_line(TOOLS_GCOV_POST_REPORT_XML, [], args)
-      @ceedling[:tool_executor].exec(command[:line], command[:options])
+    if is_utility_enabled(opts, UTILITY_NAME_GCOVR)
+      make_gcovr_reports(opts)
```

Comment:
If here we have the condition to check if enabled utility is gcovr, so other is_utility_enabled(opts, UTILITY_NAME_GCOVR) calls in internal gcovr_*() methods are not redundant?

Reply:
I removed the nested / redundant function calls. I thought about this before committing the code and left it. Since you pointed out the issue, it's worth removing to simplify the inner logic.